### PR TITLE
8337597: Macros should be defined in the global namespace in src/hotspot/share/memory/metaspace

### PR DESCRIPTION
**Pull request by caspernorrbin**
Hi everyone,

Metaspace currently contains macros that appear to be scoped within the `metaspace` namespace, but in reality, their scope is global. To avoid confusion, these macros should be moved to the global namespace.

Additionally, there are instances where macros are defined within functions for localized use. Moving these macros outside the functions would introduce unnecessary complexity. Instead, I’ve opted to `#undef` these macros immediately after their use, ensuring they remain accessible only within their respective functions.

Thanks!

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8337597](https://bugs.openjdk.org/browse/JDK-8337597): Macros should be defined in the global namespace in src/hotspot/share/memory/metaspace (**Enhancement** - P4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20610/head:pull/20610` \
`$ git checkout pull/20610`

Update a local copy of the PR: \
`$ git checkout pull/20610` \
`$ git pull https://git.openjdk.org/jdk.git pull/20610/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20610`

View PR using the GUI difftool: \
`$ git pr show -t 20610`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20610.diff">https://git.openjdk.org/jdk/pull/20610.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20610#issuecomment-2293668090)

### CODE REVIEW

**Comment by kimbarrett** [comment]
> Metaspace currently contains macros that appear to be scoped within the `metaspace` namespace, but in reality, their scope is global. To avoid confusion, these macros should be moved to the global namespace.

I see no confusion; macros are well known to not be namespace scoped. I would
prefer to keep the namespace opening at the top of the file's code.  Looking
at the JBS issue, I see this is driven by a sonarcloud warning that I think is
just bogus.  But I'll defer to @tstuefe (as the primary maintainer of
metaspace) if he prefers this style.

> Additionally, there are instances where macros are defined within functions for localized use. Moving these macros outside the functions would introduce unnecessary complexity. Instead, I’ve opted to `#undef` these macros immediately after their use, ensuring they remain accessible only within their respective functions.

The #undef of local macros is reasonable hygiene, though a bit pedantic in the
cases being changed here.

**Review by kimbarrett — CHANGES_REQUESTED** [review]
I think the sonarcloud warnings driving some of this are silly.

**Comment by tstuefe** [comment]
I agree with @kimbarrett, there is nothing to solve here. Macros are not affected by namespaces. And I prefer opening the namespaces at the start of the file, similar to include guards, since otherwise additions can creep in that are placed outside the namespace too.

**Comment by caspernorrbin** [comment]
Understandable, thank you for your comments.

**Pull request closed without merging**
